model/src/payouts.rs

```diff
@@ -79,3 +117,35 @@ impl Payouts {
         &self.short_liquidation
     }
 }
+
+struct Announcements {
+    /// The announcement which corresponds to the oracle event that
+    /// will mark the end of an epoch for a CFD.
+    settlement: Announcement,
+    /// All the intermediate oracle announcements between the start
+    /// and end of an epoch for a CFD.
+    liquidation: Vec<Announcement>,
+}
+
+impl Announcements {
+    fn new(announcements: Vec<olivia::Announcement>) -> Result<Self> {
+        let announcements = announcements
+            .into_iter()
+            .sorted_by(|a, b| a.id.cmp(&b.id))
```

Comment:
Does it make sense to sort using the complete `BitMexPriceEventId` or should we rather sort by the `timestamp` within `BitMexPriceEventId`?

Reply:
Great comment! 

Currently we have derived `PartialOrd` on `BitMexPriceEventId` and the `timestamp` field happens to be the first field, so we are effectively sorting by timestamp. Unless the timestamps are equal, in which case we would compare the `digits`. But those are always `20`, given that we always construct them using `BitMexPriceEventId::with_20_digits`.

But this is a bad approach, since someone could nonchalantly flip the order of the fields and assume that everything is fine. So I will add a patch fixing this, by manually implementing `PartialOrd` on `BitMexPriceEventId`, ignoring the `digits` field.

Reply:
See https://github.com/itchysats/itchysats/pull/2378/commits/46e0890b2c8a105ef63d70c8a2dff2410e903559.